mixnet/config.py

```diff
@@ -2,110 +2,69 @@
-    entropy: bytes
+    membership: MixMembership
+    transmission_rate_per_sec: int  # Global Transmission Rate
+    # TODO: use this to make the size of Sphinx packet constant
+    max_mix_path_length: int
 
 
 @dataclass
-class MixClientConfig:
-    emission_rate_per_min: int  # Poisson rate parameter: lambda
-    redundancy: int
-    topology: MixnetTopology
+class NodeConfig:
+    """
+    Node-specific parameters
+    """
+
+    private_key: X25519PrivateKey
+    mix_path_length: int
+    nomssip: NomssipConfig
 
 
 @dataclass
-class MixNodeConfig:
-    encryption_private_key: X25519PrivateKey
-    delay_rate_per_min: int  # Poisson rate parameter: mu
+class NomssipConfig:
+    # The target number of peers a node should maintain in its p2p network
+    peering_degree: int
 
 
 @dataclass
-class MixnetTopology:
-    # In production, this can be a 1-D array, which is accessible by indexes.
-    # Here, we use a 2-D array for readability.
-    layers: List[List[MixNodeInfo]]
-
-    def __init__(
-        self,
-        config: MixnetTopologyConfig,
-    ) -> None:
-        """
-        Build a new topology deterministically using an entropy and a given set of candidates.
-        """
-        shuffled = FisherYates.shuffle(config.mixnode_candidates, config.entropy)
-        sampled = shuffled[: config.size.num_total_mixnodes()]
+class MixMembership:
+    """
+    A list of public information of nodes in the network.
+    We assume that this list is known to all nodes in the network.
+    """
 
-        layers = []
-        for layer_id in range(config.size.num_layers):
-            start = layer_id * config.size.num_mixnodes_per_layer
-            layer = sampled[start : start + config.size.num_mixnodes_per_layer]
-            layers.append(layer)
-        self.layers = layers
+    nodes: List[NodeInfo]
 
-    def generate_route(self, mix_destination: MixNodeInfo) -> list[MixNodeInfo]:
+    def generate_route(self, length: int) -> list[NodeInfo]:
         """
         Generate a mix route for a Sphinx packet.
-        The pre-selected mix_destination is used as a last mix node in the route,
-        so that associated packets can be merged together into a original message.
-        """
-        route = [random.choice(layer) for layer in self.layers[:-1]]
-        route.append(mix_destination)
-        return route
-
-    def choose_mix_destination(self) -> MixNodeInfo:
         """
-        Choose a mix node from the last mix layer as a mix destination
-        that will reconstruct a message from Sphinx packets.
-        """
-        return random.choice(self.layers[-1])
-
-
-@dataclass
-class MixnetTopologySize:
-    num_layers: int
-    num_mixnodes_per_layer: int
-
-    def num_total_mixnodes(self) -> int:
-        return self.num_layers * self.num_mixnodes_per_layer
-
-
-# 32-byte that represents an IP address and a port of a mix node.
-NodeAddress: TypeAlias = bytes
+        return [random.choice(self.nodes) for _ in range(length)]
```

Comment:
I just modified this code to use `choices` which is much simpler and maybe efficient. 
e558ba4e6d735a5c388f0ad8f3b37bbd0473f8ac
But, I don't use `random.sample` here because I think we don't need to prevent the case which the same node is chosen multiple times. I updated the comment for this.